app/controllers/dashboard_controller.rb

```diff
@@ -10,6 +10,46 @@ def index
     end
   end
 
+  def create
+    Dashboard::Manager.add_widget_to_user(User.current, params[:widget])
+    redirect_to root_path
+  end
+
+  def destroy
+    widget = Widget.find(params[:id])
+    if widget.present? && widget.user == User.current
+      User.current.widgets.destroy(widget)
+      status = :ok
+    else
+      status = :forbidden
+    end
+    respond_to do |format|
+      format.json {render :json => params[:id], :status => status}
+    end
+  end
+
+  def reset_default
+    Dashboard::Manager.reset_user_to_default(User.current)
+    redirect_to root_path
+  end
+
+  def save_positions
+    errors = []
+    params[:widgets].each do |id, values|
+      widget = User.current.widgets.where("id = #{id}").first
+      errors << widget.errors unless widget.update_attributes(values)
+    end
+    respond_to do |format|
+      if errors.empty?
+        format.json { render :json => {}, :status => :ok }
+      else
+        format.json { render :json => errors, :status => :bad_request }
+      end
+    end
+  rescue => exception
```

Comment:
I think there are several exceptions that can occur in this block, both with bad `params` and with loading and updating the widgets.
`process_ajax_error` logs the exception.